### PR TITLE
Fixes #11937: yum is run at each cf-promises, having a massive perf cost, on redhat-like system

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/11937

### CODE REVIEW

File: tree/20_cfe_basics/cfengine/packages.cf [hunk 1]
```diff
@@ -115,7 +115,7 @@ bundle common redhat_no_locking_knowledge {
 
 
   classes:
-    redhat::
+    redhat.agent::
```
Comment (amousset) [inline tree/20_cfe_basics/cfengine/packages.cf]: could you add comment here, the purpose of this class is not obvious

**Comment by ncharles** [comment]
Commit modified

**Comment by peckpeck** [comment]
@ncharles needs manual merging

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Pull request merged**